src/ui/pages/Backup/Backup.tsx

```diff
@@ -0,0 +1,151 @@
+    const group = await getWalletGroupByAddress(wallet.address);
+    invariant(group?.id, 'Saved wallet should belong to a group');
+    await walletPort.request('updateLastBackedUp', { groupId: group.id });
+  } else {
+    await walletPort.request('updateLastBackedUp', {
+      groupId: backupContext.groupId,
+    });
+  }
+}
+
+export function Backup() {
+  const navigate = useNavigate();
+  const backupContext = useBackupContext();
+
+  const handleSessionExpired = useCallback(() => {
+    if (backupContext.appMode === 'onboarding') {
+      navigate('/onboarding/session-expired', { replace: true });
+    } else {
+      navigate(`/backup/verify-user?groupId=${backupContext.groupId}`, {
+        replace: true,
+      });
+    }
+  }, [backupContext, navigate]);
+
+  const handleSuccess = () => {
+    zeroizeAfterSubmission();
+    if (backupContext.appMode === 'onboarding') {
+      navigate('/onboarding/success');
+    } else {
+      navigate(`/backup/success?groupId=${backupContext.groupId}`);
+    }
+  };
+
+  const handleError = (error: unknown) => {
+    if (isSessionExpiredError(error)) {
+      handleSessionExpired();
+    }
+  };
+
+  const { mutate: handleSkipBackup } = useMutation({
+    mutationFn: async () => {
+      invariant(
+        backupContext.appMode === 'onboarding',
+        'Must be in onboarding appMode'
+      );
+      const wallet = await walletPort.request('getPendingWallet');
+      if (!wallet) {
+        throw new SessionExpired();
+      }
+      await accountPublicRPCPort.request('saveUserAndWallet');
+      await setCurrentAddress({ address: wallet.address });
+    },
+    onSuccess: handleSuccess,
+    onError: handleError,
+    useErrorBoundary: true,
+  });
+
+  const { mutate: handleCompleteBackup } = useMutation({
+    mutationFn: () => completeBackup(backupContext),
+    onSuccess: handleSuccess,
+    onError: handleError,
+    useErrorBoundary: true,
+  });
+
+  const searchParams =
+    backupContext.appMode === 'wallet'
+      ? `?${new URLSearchParams({ groupId: backupContext.groupId })}`
+      : '';
+
+  return (
+    <Routes>
+      <Route
+        path="/"
+        element={
+          <Navigate
+            to={{ pathname: 'info', search: searchParams }}
+            replace={true}
+          />
+        }
+      />
+      <Route
+        path="/info"
+        element={
+          <Info
+            onContinue={() =>
+              backupContext.appMode === 'onboarding'
+                ? navigate('recovery-phrase')
+                : navigate(`/backup/verify-user${searchParams}`)
+            }
+            onSkip={handleSkipBackup}
+            onExit={() => navigate('/onboarding')}
+          />
+        }
+      />
+      <Route
+        path="/verify-user"
+        element={
+          <VerifyUser
+            onSuccess={() => navigate(`recovery-phrase${searchParams}`)}
+          />
+        }
+      />
+      <Route
+        path="/recovery-phrase"
```

Comment:
I do believe that at some point the notion of "backup" will exist for both private key and seed phrase. Therefore i'd give this route a more generic name, e.g. `/secret-value` or `/reveal` or something like that

Reply:
Why not wait to make this change when it actually reflects the situation? Right now, it would look strange if the route is called `/reveal` or `/secret-value` but the `<RecoveryPhrase>` component is displayed. Or are you just avoiding changing the route in the future?

Reply:
ok, extension routes are not public, so np